release-notes-agent/src/main/java/com/epam/release/notes/agent/service/action/mail/EmailTemplateNotificationService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 EPAM Systems, Inc. (https://www.epam.com/)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.release.notes.agent.service.action.mail;
+
+import com.epam.release.notes.agent.entity.github.GitHubIssue;
+import com.epam.release.notes.agent.entity.jira.JiraIssue;
+import com.epam.release.notes.agent.entity.mail.EmailContent;
+import com.epam.release.notes.agent.entity.version.VersionStatusInfo;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+
+import java.util.List;
+import java.util.Locale;
+
+@Service
+public class EmailTemplateNotificationService implements TemplateNotificationService {
+
+    private static final String OLD_VERSION = "oldVersion";
+    private static final String NEW_VERSION = "newVersion";
+    private static final String JIRA_ISSUES = "jiraIssues";
+    private static final String GITHUB_ISSUES = "gitHubIssues";
+    private static final String EXCEPTION_MESSAGE = "This kind of version status is not handled in current method";
+
+    private final TemplateEngine templateEngine;
+    private final String emailToAdminTemplateName;
+    private final String emailToSubscribersTemplateName;
+    private final String emailToAdminTitle;
+    private final String emailToSubscribersTitle;
+
+    public EmailTemplateNotificationService(
```

Comment:
this class also should decide to whom this email will be sent, see how `EmailContent` class is configured, there is `recipients` field

Reply:
also please be accurate with null for recipients, pease fill it with empty list of null value was provided (see ListUtils.emptyIfNull functionality of Apache common lib)

Reply:
to be able to inject such `@Value` as `List<String>` use this notation of  Spring Expression Lang
`@Value("#{'${release.notes.agent.subscriber.emails}'.split(',')}")`